pkg/controllers/disruption/controller.go

```diff
@@ -127,7 +108,7 @@ func (c *Controller) Reconcile(ctx context.Context, _ reconcile.Request) (reconc
 	// Karpenter taints nodes with a karpenter.sh/disruption taint as part of the disruption process
 	// while it progresses in memory. If Karpenter restarts during a disruption action, some nodes can be left tainted.
 	// Idempotently remove this taint from candidates before continuing.
-	if err := c.requireNoScheduleTaint(ctx, false, c.cluster.Nodes()...); err != nil {
+	if err := c.Queue.RequireNoScheduleTaint(ctx, false, c.cluster.Nodes()...); err != nil {
```

Comment:
With the async queueing mechanism for waits, I don't think that we can rely on this anymore; since, we can hit this method now while the queue is still processing elements

Reply:
I'm not following, what part can we not rely on anymore?